test/Unit/Target/Php/Transpiler/UnaryOperation/UnaryOperationTranspilerTest.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/**
+ * PackageFactory.ComponentEngine - Universal View Components for PHP
+ *   Copyright (C) 2022 Contributors of PackageFactory.ComponentEngine
+ *
+ *   This program is free software: you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation, either version 3 of the License, or
+ *   (at your option) any later version.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+declare(strict_types=1);
+
+namespace PackageFactory\ComponentEngine\Test\Unit\Target\Php\Transpiler\UnaryOperation;
+
+use PackageFactory\ComponentEngine\Parser\Ast\ExpressionNode;
+use PackageFactory\ComponentEngine\Parser\Ast\UnaryOperationNode;
+use PackageFactory\ComponentEngine\Target\Php\Transpiler\UnaryOperation\UnaryOperationTranspiler;
+use PackageFactory\ComponentEngine\Test\Unit\TypeSystem\Scope\Fixtures\DummyScope;
+use PackageFactory\ComponentEngine\TypeSystem\Type\StringType\StringType;
+use PHPUnit\Framework\TestCase;
+
+final class UnaryOperationTranspilerTest extends TestCase
+{
+    /**
+     * @return array<string,mixed>
+     */
+    public function unaryOperationExamples(): array
+    {
+        return [
+            '!false' => ['!false', '(!false)'],
+            '!!false' => ['!!false', '(!(!false))'],
+            '!!!false' => ['!!!false', '(!(!(!false)))'],
+            '!1 + 1' => ['!1 + 1', '(!(1 + 1))'],
```

Comment:
```suggestion
            '!1 + 1' => ['!1 + 1', '(((!1) + 1))'],
```

The unary operation must take precedence in this case. That also means that this specific case must be moved to the `BinaryOperationTranspilerTest`, because if implemented correctly it will not fulfill the assertion:

```php
assert($node instanceof UnaryOperationNode);
```